controllers/chat/moveToAnon.js

```diff
@@ -7,6 +7,26 @@ const UsersFunction = require('../../databases/functions/users');
 const Getstream = require('../../vendor/getstream');
 const BetterSocialConstantListUtils = require('../../services/bettersocial/constantList/utils');
 
+const {
+  is_all_anon_user,
+  handle_anon_to_anon_channel_owner,
+  handle_anon_to_anon_channel_member
+} = require('../../services/bettersocial/chat/allAnonChat');
+
+const setNewStateMemberWithAnonInfo = (anon_user_info) => {
+  let user = {
+    name: `Anonymous ${anon_user_info.anon_user_info_emoji_name}`,
+    username: `Anonymous ${anon_user_info.anon_user_info_emoji_name}`
+  };
+  return {
+    anon_user_info_color_code: anon_user_info.anon_user_info_color_code,
+    anon_user_info_color_name: anon_user_info.anon_user_info_color_name,
+    anon_user_info_emoji_code: anon_user_info.anon_user_info_emoji_code,
+    anon_user_info_emoji_name: anon_user_info.anon_user_info_emoji_name,
+    user
+  };
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `setNewStateMemberWithAnonInfo` creates a new user object with anonymous information. However, it doesn't validate the input `anon_user_info`. If `anon_user_info` is null or undefined, this could lead to an error. Consider adding validation for `anon_user_info`.

```diff
const setNewStateMemberWithAnonInfo = (anon_user_info) => {
+   if (!anon_user_info) {
+     throw new Error('Invalid anonymous user info');
+   }
  let user = {
    name: `Anonymous ${anon_user_info.anon_user_info_emoji_name}`,
    username: `Anonymous ${anon_user_info.anon_user_info_emoji_name}`
  };
  return {
    anon_user_info_color_code: anon_user_info.anon_user_info_color_code,
    anon_user_info_color_name: anon_user_info.anon_user_info_color_name,
    anon_user_info_emoji_code: anon_user_info.anon_user_info_emoji_code,
    anon_user_info_emoji_name: anon_user_info.anon_user_info_emoji_name,
    user
  };
};
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->